### ISSUE TITLE
Add compatability for python3.7

### ISSUE THREAD

**Issue by nielstron** [OpShin/opshin#5]
This is currently prohibited by a slightly different setup of the python ast module

**Comment by Et9797** [comment]
Would Python 3.7 support be possible as well?

**Comment by nielstron** [comment]
Yes that should be possible too. I will run some automated tests on python3.7.

**Comment by nielstron** [comment]
It appears there are small differences in the python AST modules between different python version. Will investigate how big the changes are and if there is maybe a unifying framework (like [parso](https://parso.readthedocs.io/en/latest/)

**Comment by nielstron** [comment]
Python3.7 support will likely not come anytime soon because it appears that type hints were only introduced in python3.8
